src/networkProviders/proxyNetworkProvider.ts

```diff
@@ -129,7 +129,24 @@ export class ProxyNetworkProvider implements INetworkProvider {
     }
 
     async estimateTransactionCost(tx: Transaction): Promise<TransactionCostResponse> {
-        const transaction = prepareTransactionForBroadcasting(tx);
+        const copiedTx = Transaction.newFromPlainObject(tx.toPlainObject());
```

Comment:
This endpoint only exists on Proxy (Gateway).

Since we have this new logic written in two places, maybe, in the `ApiNetworkProvider`, simply do a call towards `backingProxyNetworkProvider`?

Reply:
Not unit tested (though, this time, they aren't strictly required).